Content.Server/Atmos/EntitySystems/AtmosphereSystem.Hotspot.cs

```diff
@@ -56,7 +61,19 @@ private void ProcessHotspot(
             if (tile.Hotspot.Bypassing)
             {
                 tile.Hotspot.State = 3;
-                // TODO ATMOS: Burn tile here
+
+                var gridUid = ent.Owner;
+                var tilePos = tile.GridIndices;
+
+                // Get prototypes of the burnt tile decals
+                var decals = _prototypeManager.EnumeratePrototypes<DecalPrototype>().Where(x => x.Tags.Contains("burnt")).Select(x => x.ID);
+
+                // Get already existing burnt decals on the tile
+                var tileDecals = _decalSystem.GetDecalsInRange(gridUid, tilePos).Select(x => decals.Contains(x.Decal.Id));
+
+                // Add a random burned decal to the tile only if there are less than 4 of them
+                if (tileDecals.Count() < 4)
+                    _decalSystem.TryAddDecal(decals.ElementAt(_random.Next(decals.Count())), new(gridUid, tilePos), out _, cleanable: true);
```

Comment:
This is too slow to use at runtime:
1. Do not use LINQ in atmos code, it is too slow.
2. Do not iterate every single prototype, dictionaries are too slow.
3. .Count() is also LINQ.